### PR TITLE
added       Added Haxe, Nexo and Neko-devel. 

### PR DESCRIPTION
**Pull request by OHNONOTAMOTH**
#### Testing the changes
- I tested the changes in this PR:**briefly**

### CODE REVIEW

File: srcpkgs/haxe/template [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+depends=""
+short_desc="High-level cross-platform programming language"
+maintainer="ohnonotamoth <dilanjohnson@outlook.com>"
+license="MIT"
+homepage="https://haxe.org/"
+distfiles="https://github.com/HaxeFoundation/haxe/archive/refs/tags/${version}.tar.gz"
+checksum=58d79c8c4e01014c00d7e431181f846c42df0324f9d97df752615efa5fcbacbd
+
+pre_build() {
+	git clone https://github.com/HaxeFoundation/haxelib.git extra/haxelib_src
```
Comment (abenson) [inline srcpkgs/haxe/template:17]: Please don't `git clone` things like this, grab it in `distfiles` and extract then move it into place.

File: srcpkgs/neko/template [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Template file for 'neko'
+pkgname=neko
+version=2.4.0
+revision=1
+#archs="i686 x86_64"
```
Comment (abenson) [inline srcpkgs/neko/template:5]: Remove any of the commented lines.

File: srcpkgs/haxe/template [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'haxe'
+pkgname=haxe
+version=4.3.5
+revision=1
+build_style=gnu-makefile
+hostmakedepends="ocaml opam wget tar pkg-config perl rsync which git m4"
+makedepends="pcre2-devel zlib-devel neko-devel mbedtls-devel"
+depends=""
```
Comment (abenson) [inline srcpkgs/haxe/template:8]: If its blank, leave it out.

File: srcpkgs/haxe/template (hunk 1 above)
Comment (OHNONOTAMOTH) [reply srcpkgs/haxe/template:17]: Haxelib has a bunch of git submodules that don't have releases, so this is the only way i could figure out how to do it.

File: srcpkgs/neko/template [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+#make_dirs="/var/log/dir 0755 root root"
+hostmakedepends="pkg-config git"
+makedepends="gc-devel openssl-devel pcre2-devel zlib-devel apache-devel libmariadbclient-devel sqlite-devel mbedtls-devel gtk+3-devel"
+depends=""
+short_desc="deprecated programming language required by Haxe"
+maintainer="ohnonotamoth <dilanjohnson@outlook.com>"
+license="MIT"
+homepage="https://nekovm.org/"
+#changelog=""
+distfiles="https://github.com/HaxeFoundation/neko/archive/refs/tags/v2-4-0.tar.gz"
```
Comment (abenson) [inline srcpkgs/neko/template:21]: - distfiles="https://github.com/HaxeFoundation/neko/archive/refs/tags/v2-4-0.tar.gz"
+ distfiles="https://github.com/HaxeFoundation/neko/archive/refs/tags/v${version//./-}.tar.gz"

Comment (abenson) [inline srcpkgs/neko/template:21]:
```suggestion
distfiles="https://github.com/HaxeFoundation/neko/archive/refs/tags/v${version//./-}.tar.gz"
```

**Comment by abenson** [comment]
You will need to rebase and squash the commits, one for each template, following our [guidelines](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#committing-your-changes) for commit messages.

**Review by OHNONOTAMOTH — COMMENTED** [review]
*empty

**Comment by abenson** [comment]
That happened because you PR'd from `master`. Create a separate branch for just these commits and then PR from that.

**Pull request closed without merging**